core/src/main/java/org/apache/accumulo/core/iterators/user/ScanYieldException.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.accumulo.core.iterators.user;
+
+import org.apache.accumulo.core.data.Key;
+
+/**
+ * This exception can be thrown at from a next or seek call on an iterator to allow other scans to get time slots. This mechanism is intended to avoid a set of
+ * scans to dominate all of the scan slots (readahead threads) and starve other scans out.
+ */
+public class ScanYieldException extends RuntimeException {
+  private final Key position;
+
+  public ScanYieldException(Key position) {
```

Comment:
> It appears that I have not adequately described the process here. Here is a scenario that might help understanding:

No, I think I understand the control flow. I'm more worried about how we pass along this "tribal knowledge" to developers who stumble onto this Exception/code-path. As we try to make SKVI more "user-friendly", I'm trying to make sure that new things which are added stand up to the level of "ready for users" and don't increase our complexity-debt :)